Sources/ConcurrencyLimitedDispatcher.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+/// A PromiseKit Dispatcher that allows no more than X simultaneous
+/// executions at once.
+
+class ConcurrencyLimitedDispatcher: Dispatcher {
```

Comment:
Nice, we can remove the `when()` that does this manually. I figure. Haven't analyzed them both.